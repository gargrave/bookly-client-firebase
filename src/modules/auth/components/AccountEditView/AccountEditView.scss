@import '../../../../styles/_views';

.accountEditView {
  @include viewWrapper;
}
