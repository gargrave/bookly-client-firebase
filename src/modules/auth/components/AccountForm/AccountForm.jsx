// @flow
import React from 'react';
import { bool, func, shape, string } from 'prop-types';

import type { Profile, ProfileErrors } from '../../../../modules/profiles/flowtypes';

import Form from '../../../../modules/common/components/Form/Form';
import InputField from '../../../../modules/common/components/InputField/InputField';

type Props = {
  disabled?: boolean,
  errors: ProfileErrors,
  onCancel: Function,
  onInputChange: Function,
  onSubmit: Function,
  profile: Profile,
  topLevelError?: string,
  submitDisabled?: boolean,
};

const profileShape = shape({
  firstName: string.isRequired,
  lastName: string.isRequired,
});

const AccountForm = ({
  disabled = false,
  errors,
  onCancel,
  onInputChange,
  onSubmit,
  profile,
  submitDisabled,
  topLevelError,
}: Props) => {
  return (
    <Form
      classes={['account-form']}
      disabled={disabled}
      onCancel={onCancel}
      onSubmit={onSubmit}
      submitDisabled={submitDisabled}
      topLevelError={topLevelError}
    >
      <InputField
        boundValue={profile.firstName}
        disabled={disabled}
        error={errors.firstName}
        label="First name"
        maxLength={255}
        name="firstName"
        onInputChange={onInputChange}
      />

      <InputField
        boundValue={profile.lastName}
        disabled={disabled}
        error={errors.lastName}
        label="Last name"
        maxLength={255}
        name="lastName"
        onInputChange={onInputChange}
      />
    </Form>
  );
};

AccountForm.propTypes = {
  profile: profileShape.isRequired,
  disabled: bool,
  errors: profileShape.isRequired,
  onCancel: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
  submitDisabled: bool,
  topLevelError: string,
};

export default AccountForm;
