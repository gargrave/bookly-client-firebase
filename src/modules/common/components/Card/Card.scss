@import '../../../../styles/_shadows';

$textColor: #6e6969;
$titleColor: #5d5656;

.card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
}

.hoverableCard:hover {
  background: #e4effc;
  cursor: pointer;
}

.header,
.title,
.text {
  margin: 0;
}

.header,
.title {
  color: $titleColor;
  font-weight: bold;
}

.header {
  font-size: 1.4rem;
}

.detailCard {
  padding: 15px;

  .title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .title + .cardText {
    font-size: 1.1rem;
  }

  .cardText + .cardText {
    margin-top: 8px;
  }
}

.listDetailCard {
  padding: 10px
}
