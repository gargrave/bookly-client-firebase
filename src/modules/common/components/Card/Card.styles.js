import styled from 'react-emotion'
import { colors, shadows } from '../../../../styles/'

export const StyledWrapper = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`
