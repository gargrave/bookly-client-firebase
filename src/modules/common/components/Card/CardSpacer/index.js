import CardSpacer from './CardSpacer'

export default CardSpacer
