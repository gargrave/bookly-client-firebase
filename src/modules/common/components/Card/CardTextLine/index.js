import CardTextLine from './CardTextLine'

export default CardTextLine
