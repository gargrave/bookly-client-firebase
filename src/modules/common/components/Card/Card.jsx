// @flow
import React from 'react'
import { any, bool, func } from 'prop-types'
import styled from 'react-emotion'

import { buildClass } from '../../../../utils/cssHelpers'

import CardHeader from './CardHeader'
import CardSpacer from './CardSpacer'
import CardTextLine from './CardTextLine'
import CardTextList from './CardTextList'

import { colors, shadows } from '../../../../styles/'

type Props = {
  children?: any,
  hoverable?: boolean,
  onClick?: Function,
}

const Styled = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
  text-align: center;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`

const Card = ({ children, hoverable = false, onClick }: Props) => {
  return (
    <Styled className={buildClass({ hoverable })} onClick={onClick}>
      {children}
    </Styled>
  )
}

Card.Header = CardHeader
Card.Spacer = CardSpacer
Card.TextLine = CardTextLine
Card.TextList = CardTextList

Card.propTypes = {
  children: any,
  hoverable: bool,
  onClick: func,
}

export default Card
