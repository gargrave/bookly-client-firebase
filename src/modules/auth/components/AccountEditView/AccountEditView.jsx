// @flow
import React from 'react';
import { bool, func, object, string } from 'prop-types';

import type { Profile } from '../../../profiles/flowtypes';

import AccountForm from '../AccountForm/AccountForm';
import Card from '../../../common/components/Card/Card';

import styles from './AccountEditView.css';

type Props = {
  disabled: boolean,
  errors: Object,
  onCancelClick: Function,
  onInputChange: Function,
  onSubmit: Function,
  profile: Profile,
  submitDisabled: boolean,
  topLevelError: string,
};

const AccountEditView = ({
  disabled,
  errors,
  onCancelClick,
  onInputChange,
  onSubmit,
  profile,
  submitDisabled = false,
  topLevelError,
}: Props) => {
  return (
    <div className={styles.accountEditView}>
      <Card
        header="Update My Account">
        <AccountForm
          disabled={disabled}
          errors={errors}
          onCancel={onCancelClick}
          onInputChange={onInputChange}
          onSubmit={onSubmit}
          profile={profile}
          submitDisabled={submitDisabled}
          topLevelError={topLevelError} />
      </Card>
    </div>
  );
};

AccountEditView.propTypes = {
  disabled: bool.isRequired,
  errors: object.isRequired,
  onCancelClick: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
  profile: object.isRequired,
  submitDisabled: bool.isRequired,
  topLevelError: string,
};

export default AccountEditView;
