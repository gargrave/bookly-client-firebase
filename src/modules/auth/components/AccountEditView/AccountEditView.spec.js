import { profileMocks } from '../../../../globals/mocks/';
import { ComponentBuilder } from '../../../../globals/utils/testHelpers';
import profileModel from '../../../profiles/models';

import AccountForm from '../AccountForm/AccountForm';
import Card from '../../../common/components/Card/Card';

import AccountEditView from './AccountEditView';

const defaultProps = {
  disabled: false,
  errors: profileModel.emptyErrors(),
  onCancelClick: jest.fn(),
  onInputChange: jest.fn(),
  onSubmit: jest.fn(),
  profile: { ...profileMocks[0] },
  submitDisabled: false,
  topLevelError: '',
};

const builder = new ComponentBuilder(
  AccountEditView,
  defaultProps,
);

describe('AccountEditView', () => {
  let component;

  test('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = builder.mountComponent();
    expect(component.find('.bookly-account-edit-view')).toHaveLength(1);
    expect(component.find(Card)).toHaveLength(1);
    expect(component.find(AccountForm)).toHaveLength(1);
  });
});
