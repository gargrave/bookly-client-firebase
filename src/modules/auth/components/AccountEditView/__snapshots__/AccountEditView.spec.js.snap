// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountEditView matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountEditView
    disabled={false}
    errors={
      Object {
        "firstName": "",
        "lastName": "",
      }
    }
    onCancelClick={[Function]}
    onInputChange={[Function]}
    onSubmit={[Function]}
    profile={
      Object {
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Larry",
        "lastName": "Finklestein",
        "owner": "oiw223h3qndlksadffl3ksdf",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    submitDisabled={false}
    topLevelError=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        header="Update My Account"
      >
        <AccountForm
          disabled={false}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          profile={
            Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          submitDisabled={false}
          topLevelError=""
        />
      </Card>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <AccountForm
          disabled={false}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          profile={
            Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          submitDisabled={false}
          topLevelError=""
        />,
        "header": "Update My Account",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "errors": Object {
            "firstName": "",
            "lastName": "",
          },
          "onCancel": [Function],
          "onInputChange": [Function],
          "onSubmit": [Function],
          "profile": Object {
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Larry",
            "lastName": "Finklestein",
            "owner": "oiw223h3qndlksadffl3ksdf",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "submitDisabled": false,
          "topLevelError": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          header="Update My Account"
        >
          <AccountForm
            disabled={false}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            profile={
              Object {
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Larry",
                "lastName": "Finklestein",
                "owner": "oiw223h3qndlksadffl3ksdf",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            submitDisabled={false}
            topLevelError=""
          />
        </Card>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <AccountForm
            disabled={false}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            profile={
              Object {
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Larry",
                "lastName": "Finklestein",
                "owner": "oiw223h3qndlksadffl3ksdf",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            submitDisabled={false}
            topLevelError=""
          />,
          "header": "Update My Account",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "errors": Object {
              "firstName": "",
              "lastName": "",
            },
            "onCancel": [Function],
            "onInputChange": [Function],
            "onSubmit": [Function],
            "profile": Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "submitDisabled": false,
            "topLevelError": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
